src/model/project.h

```diff
@@ -15,7 +15,7 @@ namespace toggl {
 
 class TOGGL_INTERNAL_EXPORT Project : public BaseModel {
  public:
-    Project() : BaseModel() {}
+    Project() : BaseModel() { EnsureGUID(); }
```

Comment:
why is EnsureGUID() needed here specifically for Project?

Reply:
In my database, some projects ended up not having GUIDs, to be able to sync them properly, I need the GUID to be there at all times. In case this GUID is loaded from the database, it gets overwritten before it can be used elsewhere in the database.

Reply:
to be honest, no, but having it there doesn't hurt. guid should be there by default for all entities but unless we're able to reproduce it for other entities, i don't think it makes sense to add it elsewhere now

Reply:
I can remove this line. However, the fact if the TE is synced or not is determined by if it has an ID or not. Having an extra GUID there right away doesn't change current behavior.